### ISSUE TITLE
Fix logic for reaping apps based on total app limits

### ISSUE THREAD

**Issue by schneems** [heroku/hatchet#90]
When I originally wrote hatchet teams were not a thing and personal accounts were limited to 100 apps. So the default limit says "do not go over 100 apps" . However this logic is severely flawed since an account can have more than 100 apps returned from the API call via `app.list` of the `platform-api` client. For instance:

```
Warning: Reached Heroku app limit of 100. hatchet app count: 1, all app count: 224
```

So what happens in this scenario is that Hatchet thinks it only has room to create and use a single app. Due to the distributed nature of Hatchet (running in parallel across multiple processes) each process is trying to destroy and create the same app which ends up creating a destroy loop since none of them let it live long enough to finish a test. This causes some pretty catastrophic test failures even in this scenario the account above has capacity to consume/create more apps.

**Comment by schneems** [comment]
We could grep the error message:

```
{"id":"invalid_params","message":"You've reached your app limit of 100 apps. Delete some apps or open a ticket at https://help.heroku.com/new/increase-limits to have this limit increased."} 
```

But...i'm not sure if the `platform-api` gem will expose it. I'll have to dig in more

**Comment by schneems** [comment]
Apparently there's two different outputs we need to account for:

```
You've reached the limit of 100 apps
```

**Comment by edmorley** [comment]
@schneems Can this be closed now that #97 has merged? :-)
